IPython/html/static/tree/js/notebooklist.js

```diff
@@ -229,34 +234,84 @@ define([
         }
         for (var i=0; i<len; i++) {
             model = list.content[i];
-            item = this.new_item(i+offset);
+            item = this.new_item(i+offset, true);
             this.add_link(model, item);
         }
         // Trigger an event when we've finished drawing the notebook list.
         events.trigger('draw_notebook_list.NotebookList');
+        this._selection_changed();
     };
 
 
-    NotebookList.prototype.new_item = function (index) {
-        var item = $('<div/>').addClass("list_item").addClass("row");
-        // item.addClass('list_item ui-widget ui-widget-content ui-helper-clearfix');
-        // item.css('border-top-style','none');
-        item.append($("<div/>").addClass("col-md-12").append(
-            $('<i/>').addClass('item_icon')
-        ).append(
-            $("<a/>").addClass("item_link").append(
-                $("<span/>").addClass("item_name")
-            )
-        ).append(
-            $('<div/>').addClass("item_buttons  pull-right")
-        ));
+    /**
+     * Creates a new item.
+     * @param  {integer} index
+     * @param  {boolean} [selectable] - tristate, undefined: don't draw checkbox,
```

Comment:
Okay. I think that suggests we might be using one function for too many things, but it's okay, for now. I like the result, in general.